vg/vgeps/vgeps.go

```diff
@@ -176,17 +178,106 @@ func (e *Canvas) trace(path vg.Path) {
 	}
 }
 
-func (e *Canvas) FillString(fnt vg.Font, x, y vg.Length, str string) {
+func (e *Canvas) FillString(fnt vg.Font, pt vg.Point, str string) {
 	if e.cur().font != fnt.Name() || e.cur().fsize != fnt.Size {
 		e.cur().font = fnt.Name()
 		e.cur().fsize = fnt.Size
 		fmt.Fprintf(e.buf, "/%s findfont %.*g scalefont setfont\n",
 			fnt.Name(), pr, fnt.Size)
 	}
-	fmt.Fprintf(e.buf, "%.*g %.*g moveto\n", pr, x.Dots(DPI), pr, y.Dots(DPI))
+	fmt.Fprintf(e.buf, "%.*g %.*g moveto\n", pr, pt.X.Dots(DPI), pr, pt.Y.Dots(DPI))
 	fmt.Fprintf(e.buf, "(%s) show\n", str)
 }
 
+// DrawImage implements the vg.Canvas.DrawImage method.
+func (c *Canvas) DrawImage(rect vg.Rectangle, img image.Image) {
+	var (
+		width   = rect.Size().X.Dots(DPI)
+		height  = rect.Size().Y.Dots(DPI)
+		iwidth  = img.Bounds().Dx()
+		iheight = img.Bounds().Dy()
+	)
+	c.Push()
+	fmt.Fprintf(c.buf, "\n%%%% Image data\n")
+	c.Translate(rect.Min)
+	fmt.Fprintf(c.buf, "%.*g %.*g scale     %% scale pixels to points\n", pr, width, pr, height)
+	fmt.Fprintf(c.buf, "%d %d 8 [%d 0 0 %d 0 0]     ", iwidth, iheight, iwidth, iheight)
+	fmt.Fprintf(c.buf, "%%%% width height colordepth transform\n")
+	fmt.Fprintf(c.buf, "/datasource currentfile /ASCII85Decode filter /RunLengthDecode filter def\n")
+	fmt.Fprintf(c.buf, "/datastring %d string def   ", iwidth*3)
+	fmt.Fprintf(c.buf, "%% %d = width * color components\n", iwidth*3)
+	fmt.Fprintf(c.buf, "{datasource datastring readstring pop}\n")
+	fmt.Fprintf(c.buf, "false       %%%% false == single data source (rgb)\n")
+	fmt.Fprintf(c.buf, "3   %%%% number of color components\n")
```

Comment:
It's been forever since I've looked at any of this. Do you know if we can do 4 components and add an alpha?